File: frontend/types/listing.ts
export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
  hasMore: boolean;
}

export interface Listing {
  id: string;
  title: string;
  description: string;
  address: string;
  price: number;
  startDate: string;
  endDate: string;
  isAvailable: boolean;
  createdAt: string;
  ownerId: string;
  latitude?: number | null;
  longitude?: number | null;
  isFavorited?: boolean | null;
  imageUrls: string[];
  ownerName: string;
  ownerProfileImageUrl?: string | null;
  ownerAverageRating?: number | null;
  ownerTotalRatings: number;
}

export interface CreateListingRequest {
  title: string;
  description: string;
  address: string;
  price: number;
  startDate: string;
  endDate: string;
  latitude?: number;
  longitude?: number;
}

export interface UpdateListingRequest extends CreateListingRequest {
  isAvailable: boolean;
}

File: backend/Listings/Listing.cs
using System.ComponentModel.DataAnnotations.Schema;
using OpenSpot.Users.Models;

namespace OpenSpot.Listings.Models
{
    public class Listing
    {
        public Guid Id { get; set; }

        public string Title { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public string Address { get; set; } = string.Empty;

        public int Price { get; set; }

        public DateOnly StartDate { get; set; }

        public DateOnly EndDate { get; set; }

        public bool IsAvailable { get; set; }

        public DateTime CreatedAt { get; set; }

        public string OwnerId { get; set; } = string.Empty;

        [ForeignKey("OwnerId")]
        public User Owner { get; set; } = null!;

        public double? Latitude { get; set; }

        public double? Longitude { get; set; }

        public ICollection<ListingImage> Images { get; set; } = new List<ListingImage>();
    }
}